lib/app/sort_filter/sort_filter.dart

```diff
@@ -11,14 +12,61 @@ typedef SortFilterChangeCallback<T> = void Function(
 
 @immutable
 class SortFilter<T> {
-  const SortFilter({
-    this.sortOptions = const {},
+  SortFilter({
+    this.sorters = const {},
+    @required this.defaultSorter,
+    this.defaultSortOrder = SortOrder.ascending,
     this.filters = const {},
-  })  : assert(sortOptions != null),
+  })  : assert(sorters != null),
+        assert(sorters.isNotEmpty),
+        assert(defaultSorter != null),
+        assert(defaultSortOrder != null),
         assert(filters != null);
 
-  final Map<String, Sorter<T>> sortOptions;
+  final Map<String, Sorter<T>> sorters;
+  final String defaultSorter;
+  final SortOrder defaultSortOrder;
   final Map<String, Filter> filters;
+
+  SortFilterSelection<T> get defaultSelection {
+    return SortFilterSelection(
+      config: this,
+      sortSelectionKey: defaultSorter,
+      sortOrder: defaultSortOrder,
+      filterSelections: {
+        for (final entry in filters.entries)
+          entry.key: entry.value.defaultSelection,
+      },
+    );
+  }
+
+  /// Parses a query string generated by the web client, e.g.
+  /// https://schul-cloud.org/homework/#?dueDateFrom=2020-03-09&dueDateTo=2020-03-27&private=true&publicSubmissions=false&sort=updatedAt&sortorder=1&teamSubmissions=true
+  SortFilterSelection<T> tryParseWebQuery(Map<String, String> query) {
+    if (query.isEmpty) {
+      return defaultSelection;
+    }
+
+    return SortFilterSelection(
+      config: this,
+      sortSelectionKey: _tryParseWebQuerySorter(query),
+      sortOrder: SortOrderUtils.tryParseWebQuery(query) ?? defaultSortOrder,
+      filterSelections: {
+        for (final entry in filters.entries)
+          entry.key: entry.value.tryParseWebQuerySorter(query, entry.key),
+      },
+    );
+  }
+
+  String _tryParseWebQuerySorter(Map<String, String> query) {
+    final key = query['sort'];
+    return sorters.entries
+        .firstWhere(
+          (option) => key == (option.value.webQueryKey ?? option.key),
+          orElse: () => MapEntry(defaultSorter, null),
+        )
+        .key;
```

Comment:
I believe this is more readable:
```suggestion
    return sorters.entries
        .firstWhere(
          (option) => key == (option.value.webQueryKey ?? option.key),
          orElse: () => null,
        )
        ?.key ?? defaultSorter;
```